src/sif/pedestriancost.cc

```diff
@@ -484,6 +517,10 @@ class PedestrianCost : public DynamicCost {
     c.cost +=
         maneuver_penalty_ * (!edge->link() && edge->use() != Use::kEgressConnection &&
                              edge->use() != Use::kPlatformConnection && !edge->name_consistency(idx));
+    c.cost += living_street_penalty_ *
+              (edge->use() == baldr::Use::kLivingStreet && pred->use() != baldr::Use::kLivingStreet);
+    c.cost +=
+        track_penalty_ * (edge->use() == baldr::Use::kTrack && pred->use() != baldr::Use::kTrack);
```

Comment:
As I see this is the only change around logic itself and everything else is renaming, right? Maybe we can leave naming as it is and apply only these cost-specific changes?

Reply:
1. Fixed incorrect method overriding. (35925a3bef82018e39157959485f26f77cf88bf6)
@merkispavel There is the incorrect function overriding!
Example:
```
DynamicCost *ptr = new PedestrianCost(...)
ptr->base_transition_cost(...)  
```
DynamicCost::base_transition_cost was called before these changes, and PedestrianCost::base_transition_cost after.

2. Renamed base_transition_cost -> transition_cost. (174a4ac541dc10db0930dd74ad96a1e19a71d687)
@mandeepsandhu may be we should add prefix `base_` in each base class method? It's always the common something applicable for all derived classes:)
Why is PedestrianCost::base_transition_cost named as base_transition_cost?

Why shouldn't these changes be merged into the master?

Reply:
Hmmm, to me 'base' in base_transition_cost sounded more like 'common', something that we start with. And having it as a function in some base class is only a way to implement it.

Also, with the naming suggested in this pr we get methods named `Cost::TransitionCost` and `Cost::transition_cost` at the same time, which may be more confusing

Reply:
> Also, with the naming suggested in this pr we get methods named Cost::TransitionCost and Cost::transition_cost at the same time, which may be more confusing

agreed, not the best combo 😕

Reply:
> Fixed incorrect method overriding. (35925a3)
@merkispavel There is the incorrect function overriding!
Example:
DynamicCost *ptr = new PedestrianCost(...)
ptr->base_transition_cost(...) 

As far as I see `base_transition_cost` is a **protected** method so I'm not sure the listed example is possible

Reply:
@dnesbitt61 i believe you are the original author here. can you let us know your thoughts?

Reply:
I likely only ever intended this to be called from within the derived class's TransitionCost methods.